test/integration/targets/git/tasks/main.yml

```diff
@@ -291,6 +292,75 @@
     that:
       - 'git_result.stdout == "2cfde3668b8bb10fbe2b9d5cec486025ad8cc51b"'
 
+
+# Test that a forced shallow checkout referincing branch only always fetches latest head
+
+- name: clear checkout_dir
+  file: state=absent path={{ item }}
+  with_items:
+    - "{{ checkout_dir }}"
+    - "{{ checkout_dir }}.copy"
+
+- name: checkout example repo
+  git: repo={{ repo_format1 }} dest={{ checkout_dir }}
+
+- name: clone example repo locally
+  git: repo={{ checkout_dir }} dest={{checkout_dir}}.copy
+
+- name: create branch in original
+  command: git checkout -b test/branch chdir="{{checkout_dir}}"
+
+- name: get commit for HEAD on new branch
+  command: git rev-parse HEAD chdir="{{checkout_dir}}.copy"
+  register: originaltip0
+
+- name: shallow force checkout new branch in copy
+  git: repo={{ checkout_dir }} dest={{checkout_dir}}.copy version=test/branch refspec=refs/heads/test/branch depth=1 force=yes
+
+- name: create new commit in original
+  command: git commit --allow-empty -am . chdir="{{checkout_dir}}"
```

Comment:
I would find `echo a > b; git add b; git commit -m 'add b'` more readable than the `--allow-empty` construct.